server/src/main/java/org/opensearch/index/compositeindex/datacube/MetricStat.java

```diff
@@ -10,29 +10,59 @@
 
 import org.opensearch.common.annotation.ExperimentalApi;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
 /**
  * Supported metric types for composite index
  *
  * @opensearch.experimental
  */
 @ExperimentalApi
 public enum MetricStat {
-    COUNT("count"),
-    AVG("avg"),
-    SUM("sum"),
-    MIN("min"),
-    MAX("max");
+    COUNT("count", null),
```

Comment:
should it be null or empty list?

Reply:
Didn't want to initialize empty lists , hence went with array. Will keep this comment open if any other reviewers have concern as well, then will change it.